CHANGELOG.md

```diff
@@ -6,6 +6,56 @@ command line for details.
 
 ## [Unreleased]
 
+## 5.0
+
+### 5.0.0 (prerelease) - 2025-05
+
+5.0.0 is a major revision, though there are no feature changes in this package.
+_Hopefully_ the only change folks should see is improved stability and resource usage.
+
+Major changes:
+
+- The long-abandoned `http-proxy` dependency,
+  which implements most of what configurable-http-proxy actually does,
+  has been replaced with [`http-proxy-3`](https://www.npmjs.com/package/http-proxy-3), a fully compatible, updated and modern fork.
+  This should hopefully fix a lot of resource exhaustion and other issues folks have been seeing.
+  Thank you [@williamstein](https://github.com/williamstein)!
+- The minimum nodejs runtime is now node 18 (up from 10).
+  This won't affect most users, as node 18 has already reached end-of-life at time of release,
+  but folks who get node from their distro package manager (e.g. `apt` or `yum`) often lag behind.
+  Node 18 is in Ubuntu 24.04 LTS and Debian 12/Stable.
+- CHP now uses ESM instead of CommonJS (`import` instead of `require`).
+  This shouldn't have any visible effect,
+  except for anyone calling `require("configurable-http-proxy")` on node 18,
+  which doesn't support CommonJS `require` of an ES Module.
+  `require("configurable-http-proxy")` works as before on node >=20.
+- nodejs runtime in the quay.io/jupyterhub/configurable-http-proxy image is increased to node 24 from node 20.
+
+([full changelog](https://github.com/jupyterhub/configurable-http-proxy/compare/4.6.3...5.0.0b1))
+
+#### API and Breaking Changes
```

Comment:
I have used the breaking label as a reminder on what to write about manually when generating the changelog with github-activity.

With a clear section about it already, and the PRs.listed under maintenance already, i figure this heading can be removed